pkg/rbd/rbd_util.go

```diff
@@ -109,36 +116,98 @@ type rbdSnapshot struct {
 
 var (
 	supportedFeatures = sets.NewString("layering")
+
+	// large interval and timeout, it should be longer than the maximum
+	// time an operation can take (until refcounting of the connections is
+	// available)
+	connPool = util.NewConnPool(15*time.Minute, 10*time.Minute)
 )
 
 // createImage creates a new ceph image with provision and volume options.
 func createImage(ctx context.Context, pOpts *rbdVolume, volSz int64, cr *util.Credentials) error {
-	var output []byte
-
-	image := pOpts.RbdImageName
 	volSzMiB := fmt.Sprintf("%dM", volSz)
+	options := librbd.NewRbdImageOptions()
+
+	imageFormat, err := strconv.Atoi(rbdImageFormat2)
+	if err != nil {
+		return errors.Wrapf(err, "[BUG] rbdImageFormat2 should be convertable to integer")
+	}
+	if pOpts.ImageFormat != "" {
+		imageFormat, err = strconv.Atoi(pOpts.ImageFormat)
```

Comment:
this should be done at input validation isn't it? means before reserving rados keys

Reply:
I am not sure what you mean with this. Can you explain a little more?

Reply:
Image format 1 is not supported by librbd anymore, so it will always be image format 2.

Reply:
@Madhu-1 @humblec are you fine with removing ImageFormat references? I do not see it configurable or used elsewhere. If there is any historic need, please let me know.

Reply:
References to ImageFormat have been removed now with an extra commit.